pkg/ansible/controller/reconcile.go

```diff
@@ -225,27 +223,21 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		}
 	}
 	if r.ManageStatus {
-		err = r.markDone(u, request.NamespacedName, statusEvent, failureMessages)
-		if err != nil {
-			logger.Error(err, "Failed to mark status done")
-		}
+		merr := r.markDone(u, request.NamespacedName, statusEvent, failureMessages)
+		checkAPIErr(merr, logger)
 	}
-	return reconcileResult, err
+	return reconcileResult, nil
```

Comment:
Should not better keep the err, since something may not return the error on top and then the top and then, the reconcile will not be trigged when should?